src/diffusers/loaders.py

```diff
@@ -2276,6 +2436,93 @@ def enable_lora_for_text_encoder(self, text_encoder: Optional[PreTrainedModel] =
             raise ValueError("Text Encoder not found.")
         set_adapter_layers(self.text_encoder, enabled=True)
 
+    def set_adapters(
+        self,
+        adapter_names: Union[List[str], str],
+        unet_weights: List[float] = None,
+        te_weights: List[float] = None,
+        te2_weights: List[float] = None,
+    ):
+        # Handle the UNET
+        self.unet.set_adapters(adapter_names, unet_weights)
+
+        # Handle the Text Encoder
+        if hasattr(self, "text_encoder"):
+            self.set_adapters_for_text_encoder(adapter_names, self.text_encoder, te_weights)
+        if hasattr(self, "text_encoder_2"):
+            self.set_adapters_for_text_encoder(adapter_names, self.text_encoder_2, te2_weights)
+
+    def disable_lora(self):
+        if not self.use_peft_backend:
+            raise ValueError("PEFT backend is required for this method.")
+
+        # Disable unet adapters
+        self.unet.disable_lora()
+
+        # Disable text encoder adapters
+        if hasattr(self, "text_encoder"):
+            self.disable_lora_for_text_encoder(self.text_encoder)
+        if hasattr(self, "text_encoder_2"):
+            self.disable_lora_for_text_encoder(self.text_encoder_2)
+
+    def enable_lora(self):
+        if not self.use_peft_backend:
+            raise ValueError("PEFT backend is required for this method.")
+
+        # Enable unet adapters
+        self.unet.enable_lora()
+
+        # Enable text encoder adapters
+        if hasattr(self, "text_encoder"):
+            self.enable_lora_for_text_encoder(self.text_encoder)
+        if hasattr(self, "text_encoder_2"):
+            self.enable_lora_for_text_encoder(self.text_encoder_2)
+
+    def set_lora_device(self, adapter_names: List[str], device: Union[torch.device, str, int]) -> None:
```

Comment:
Should this be a `peft` utility?

Reply:
Yes I think that it can be! 
But I prefer to have it inside `LoraLoaderMixin` so that the API is slightly smoother --> instead of importing that method users would just have to call `pipe.set_lora_device`. What do you think?

Reply:
Could you walk me through the use case you have on mind?

Reply:
Yes sure, so there is a test that has been specifically designed for that: https://github.com/younesbelkada/diffusers/blob/peft-part-2/tests/lora/test_lora_layers_peft.py#L875 please have a look at it

Reply:
Okay, now I understand!

A couple more comments here:

* Could we use `torch.float16` here? https://github.com/younesbelkada/diffusers/blob/3708ed9cb2cefd362ecdc12739520ec5b1894a71/tests/lora/test_lora_layers_peft.py#L879
* Could we add some comments to explain what we're testing for before https://github.com/younesbelkada/diffusers/blob/3708ed9cb2cefd362ecdc12739520ec5b1894a71/tests/lora/test_lora_layers_peft.py#L894 ?
* Remove https://github.com/younesbelkada/diffusers/blob/3708ed9cb2cefd362ecdc12739520ec5b1894a71/tests/lora/test_lora_layers_peft.py#L907.

Reply:
Sure yes !

Reply:
Done in 323612b